extensions/blocks/donations/editor.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { getBlockType, registerBlockType, unregisterBlockType } from '@wordpress/blocks';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import registerJetpackBlock from '../../shared/register-jetpack-block';
+import { name, settings } from '.';
+
+/**
+ * Appends a "paid" tag to the Donations block title if site requires an upgrade.
+ */
+const addPaidBlockFlags = async () => {
```

Comment:
I would recommend relying on the existing Jetpack tools to handle this. You can look at the Calendly check for an example, or the Pay With PayPal block.

Reply:
Do we have any existing issue with the paid labels? I don't see the flag on the Calendly block after setting a JN sites on a free plan:

<img width="423" alt="Screen Shot 2020-07-23 at 10 48 36" src="https://user-images.githubusercontent.com/1233880/88268119-3ace0480-ccd2-11ea-9e71-1d79d367f05b.png">

Reply:
The Calendly block is free for self-hosted users; we only apply the plan restriction on WordPress.com Simple sites.